skimage/draw/draw.py

```diff
@@ -693,3 +693,57 @@ def bezier_curve(r0, c0, r1, c1, r2, c2, weight, shape=None):
            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=uint8)
     """
     return _bezier_curve(r0, c0, r1, c1, r2, c2, weight, shape)
+
+
+def rectangle(origin, extent, shape=None):
+    """Generate coordinates of pixels within an nd-rectangle.
+
+    Parameters
+    ----------
+    origin : tuple
+        origin of the rectangle, (page,row,column,channel)
+    extent : tuple
+        extent (num_pages,num_rows,num_cols,num_chanels..) of the rectangle.
+    shape : tuple, optional
+        Image shape which is used to determine the maximum extent of output
+        pixel coordinates. This is useful for rectangles that exceed the image
+        size. If None, the bounds are determined from the bounds of
+        the rectangle.
+
+    Returns
+    -------
+    coords : (M, n) ndarray of int with the
+        Pixel coordinates of the nd rectangle.
+        May be used to directly index into an array,
+        e.g. for a 2d rectangle
+        ``img[rr, cc] = 1``.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skimage.draw import rectangle
+    >>> img = np.zeros((5, 5), dtype=np.uint8)
+    >>> origin = (1, 1)
+    >>> extent = (2, 2)
+    >>> rr, cc = rectangle(origin, extent, img.shape)
+    >>> img[rr, cc] = 1
+    >>> img
+    array([[0, 0, 0, 0, 0],
+           [0, 1, 1, 0, 0],
+           [0, 1, 1, 0, 0],
+           [0, 0, 0, 0, 0],
+           [0, 0, 0, 0, 0]], dtype=uint8)
+
+    """
+    if not(len(origin) == len(extent)):
+        raise ValueError("origin and extent must have the same\
+                          number of dimensions")
```

Comment:
This doesn't do what you think it does:

```python
>>> a = ("origin and extent must have same\
...       number of dimensions")
>>> a
'origin and extent must have same      number of dimensions'
```

Instead, in Python within parentheses, you can simply include multiple strings with only whitespace between them, and they will get concatenated:

```python
>>> b = ("origin and extent must have same "
...      "number of dimensions")
>>> b
'origin and extent must have same number of dimensions'
```